repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/glossary/TermPreProcessor.java

```diff
@@ -120,12 +147,109 @@ private void processUpdateTerm(AtlasEntity entity, AtlasVertex vertex) throws At
             throw new AtlasBaseException(AtlasErrorCode.ACHOR_UPDATION_NOT_SUPPORTED);
         }
 
-        String vertexQnName = vertex.getProperty(QUALIFIED_NAME, String.class);
+        String vertexQName = vertex.getProperty(QUALIFIED_NAME, String.class);
+
+        entity.setAttribute(QUALIFIED_NAME, vertexQName);
+
+        String termGuid = GraphHelper.getGuid(vertex);
+
+        if(!termName.equals(vertexName) && checkForEntityAssociationToTerm(vertexQName)){
+            try {
+                if (taskManagement != null && DEFERRED_ACTION_ENABLED) {
+                    createAndQueueTask(MEANINGS_TEXT_UPDATE, termName, vertexQName, vertex, ELASTICSEARCH_PAGINATION_OFFSET);
```

Comment:
ELASTICSEARCH_PAGINATION_OFFSET is a static constant, no need to save it with task itself
It can be accessed from anywhere as needed